phy/cluster/manual/store.py

```diff
@@ -70,85 +79,68 @@ def __init__(self, directory):
+            assert isinstance(value, np.ndarray)
+            with open(self._cluster_path(cluster, key), mode) as f:
+                value.tofile(f)
+
+    def _get(self, cluster, key, dtype=None, shape=None):
         # The cluster doesn't exist: return None for all keys.
-        if not self._cluster_file_exists(cluster):
-            if keys is None:
-                return {}
-            elif isinstance(keys, string_types):
-                return None
-            elif isinstance(keys, list):
-                return {key: None for key in keys}
-            else:
-                raise ValueError(keys)
-        # Create the output dictionary.
+        if not self._cluster_file_exists(cluster, key):
+            return None
+        else:
+            with open(self._cluster_path(cluster, key), 'rb') as f:
+                return _load_ndarray(f, dtype=dtype, shape=shape)
+
+    def load(self, cluster, keys, dtype=None, shape=None):
+        """Load cluster-related data. Return a file handle, to be used
+        with np.fromfile() once the dtype and shape are known."""
+        assert keys is not None
+        if isinstance(keys, string_types):
+            return self._get(cluster, keys, dtype=dtype, shape=shape)
+        assert isinstance(keys, list)
         out = {}
-        # Open the cluster file in read mode.
-        with self._cluster_file(cluster, 'r') as f:
-            # If a single key is requested, return the value.
-            if isinstance(keys, string_types):
-                return self._get(f, keys)
-            # All keys are requested if None.
-            if keys is None:
-                keys = f.datasets()
-            assert isinstance(keys, (list, tuple))
-            # Fetch the values for all requested keys.
-            for key in keys:
-                out[key] = self._get(f, key)
+        for key in keys:
+            out[key] = self._get(cluster, key, dtype=dtype, shape=shape)
         return out
 
     @property
-    def clusters(self):
-        """List of cluster ids in the store."""
+    def files(self):
+        """List of files present in the directory."""
         if not op.exists(self._directory):
             return []
-        files = os.listdir(self._directory)
-        clusters = [int(op.splitext(file)[0]) for file in files]
+        return sorted(os.listdir(self._directory))
+
+    @property
+    def clusters(self):
+        """List of cluster ids in the store."""
+        files = self.files
+        clusters = set([int(op.splitext(file)[0]) for file in files])
         return sorted(clusters)
 
     def delete(self, clusters):
         """Delete some clusters from the store."""
-        for cluster in clusters:
-            if self._cluster_file_exists(cluster):
-                os.remove(self._cluster_path(cluster))
+        for file in self.files:
+            os.remove(op.join(self._directory, file))
 
     def clear(self):
         """Clear the store completely by deleting all clusters."""
```

Comment:
this will only delete all the clusters in `self.clusters` - depending how this is called, it might not delete clusters which no longer exist? i.e. if you merge 123 with 234 and create a new cluster 235, then `self.clusters` won't contain 123 or 234 any more?